docusaurus/docs/iOS/swiftui/components/message-composer.md

```diff
@@ -0,0 +1,395 @@
+
+In order to do this, you need to implement the `makeLeadingComposerView`, which receives a binding of the `PickerTypeState`. Having the `PickerTypeState` as a parameter allows you to control the visibility of the attachment picker view. The `PickerTypeState` has two states - expanded and collapsed. If the state is collapsed, the composer is in the minimal mode (only the text input and leading and trailing areas are shown). If the enum state is expanded, it has associated value with it, which is of type `AttachmentPickerType`. This defines the type of picker which is currently displayed in the attachment picker view. The possible states are `none` (nothing is selected), `media` (media picker is selected), `giphy` (giphy commands picker is shown) and custom (for your own custom pickers).
+
+Here's an example on how to provide a view for the leading composer view: 
+
+```swift
+public func makeLeadingComposerView(
+        state: Binding<PickerTypeState>
+) -> some View {
+    AttachmentPickerTypeView(pickerTypeState: state)
+}
+```
+
+## Customizing the AttachmentPickerTypeView
+
+The `AttachmentPickerTypeView` comes with a lot of functionalities, in terms of image and video picker, file picker and selecting media from the camera. While you can just swap this component if it doesn't fit your needs (like shown above), it's also possible to extend it with additional message attachment types to fit your needs. 
+
+For example, let's say we want to add additional contacts picker, which will allow us to send contacts via the chat. We also want to keep the existing functionalities, therefore we will explore ways to customize the existing component.
+
+There are few things we need to do in order to accomplish this: 
+- introduce new type of payload for contacts
+- create new contact attachment type component
+- enable the new contact component to be selectable from the attachment types picker
+- implement the UI for previewing the selected item (contact) in the message composer
+- update the message resolving logic of the message list to include the contact payload
+- provide UI for the contacts attachment in the message list
+
+Let's explore these steps in more details
+
+### Contact Payload
+
+First, we need to create a new `AttachmentType` for contacts, and define its payload.
+
+```swift
+extension AttachmentType {
+    static let contact = Self(rawValue: "contact")
+}
+
+struct ContactAttachmentPayload: AttachmentPayload {
+    static let type: AttachmentType = .contact
+
+    let name: String
+    let phoneNumber: String
+}
+```
+
+Since we will go through the attachments in a scrollable list, we also need to conform the payload to the `Identifiable` protocol. For an id, it's enough to use a key that's combination of the name and the phone number.
```

Comment:
```suggestion
Since we will go through the attachments in a scrollable list, we also need to conform the payload to the `Identifiable` protocol. For an id, it's enough to use a key that's a combination of the name and the phone number.
```